### PR TITLE
Refactored ACNVModeledSegment to be a subclass of ModeledSegment.  Pu…

### LINKED ISSUE #327: Add GATK CNV seg file output for Allelic CNV
GATK CNV is log2CR 
ACNV is CR
ReCapSeg is log2CR

### PR DESCRIPTION
**Pull request by LeeTL1220**
…t in functionality to write a seg file in the same format that comes out of GATK CNV.

Closes #327

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java [hunk 1]
```diff
@@ -9,44 +8,64 @@
...
 
-    private final SimpleInterval interval;
     private final PosteriorSummary segmentMeanPosteriorSummary;
     private final PosteriorSummary minorAlleleFractionPosteriorSummary;
 
     public ACNVModeledSegment(final SimpleInterval interval,
                               final PosteriorSummary segmentMeanPosteriorSummary,
                               final PosteriorSummary minorAlleleFractionPosteriorSummary) {
-        this.interval = interval;
+        super(interval, ModeledSegment.NO_CALL, 0, segmentMeanPosteriorSummary.mean());
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: note that `mean()` -> `center()` with #330

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java [hunk 2]
```diff
@@ -9,44 +8,64 @@
...
 
-    public SimpleInterval getInterval() {
-        return interval;
-    }
 
-    public PosteriorSummary getSegmentMeanPosteriorSummary() {
-        return segmentMeanPosteriorSummary;
+    /**
+     *  Get segment mean in log2 space
+     * @return
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: i'd remove the return tag if not used

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java [hunk 3]
```diff
@@ -9,44 +8,64 @@
...
+    public double getSegmentMean() {
+        return segmentMeanPosteriorSummary.mean();
     }
 
-    public PosteriorSummary getMinorAlleleFractionPosteriorSummary() {
-        return minorAlleleFractionPosteriorSummary;
+    /**
+     * Get the segment mean in non-logged space
+     *
+     * @return
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: same here

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java [hunk 4]
```diff
@@ -9,44 +8,64 @@
...
+     */
+    public double getSegmentMeanInCRSpace() {
+        return Math.pow(2, mean);
     }
 
+    /**
+     * Not supported in this class
+     *
+     * @param segmentMean
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: same w/ param tag here

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java [hunk 5]
```diff
@@ -26,14 +26,28 @@ public ModeledSegment(final SimpleInterval interval, final long targetCount, fin
         this(interval, NO_CALL, targetCount, segmentMean);
     }
 
+    /**
+     *  Get segment mean in log2 space
+     * @return
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: same

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java [hunk 6]
```diff
@@ -26,14 +26,28 @@ public ModeledSegment(final SimpleInterval interval, final long targetCount, fin
...
+     */
     public double getSegmentMean() {
         return mean;
     }
 
+    /**
+     * Set the segment mean in log2 space
+     *
+     * @param segmentMean
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: same

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java [hunk 7]
```diff
@@ -26,14 +26,28 @@ public ModeledSegment(final SimpleInterval interval, final long targetCount, fin
...
+     */
     public void setSegmentMean(final double segmentMean) {
         this.mean = segmentMean;
     }
 
+    /**
+     * Get the segment mean in non-logged space
+     *
+     * @return
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: same

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java [hunk 8]
```diff
@@ -26,14 +26,28 @@ public ModeledSegment(final SimpleInterval interval, final long targetCount, fin
         this(interval, NO_CALL, targetCount, segmentMean);
     }
 
+    /**
+     *  Get segment mean in log2 space
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: not in this diff, but i noticed that the constructor in this class takes parameter `segmentMeanInLogCR`.  maybe rename to `segmentMeanInLog2CR` to be absolutely clear?

**Comment by samuelklee** [comment]
OK, minor javadoc comments, but looks good to me otherwise.  also note that i've opened #344.  please wait until #330 is in before merging.

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java (hunk 2 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: Wrote a quick comment

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java (hunk 3 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: Wrote a quick comment

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java (hunk 4 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ACNVModeledSegment.java]: Wrote a quick comment

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java (hunk 5 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: Wrote a quick comment

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java (hunk 6 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: Wrote a quick comment

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java (hunk 7 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: Wrote a quick comment

File: src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java (hunk 8 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/ModeledSegment.java]: Done and corrected bug.

**Pull request merged**